src/main/java/com/teragrep/lsh_01/conversion/JsonPayload.java

```diff
@@ -0,0 +1,71 @@
+/*
+  logstash-http-input to syslog bridge
+  Copyright 2024 Suomen Kanuuna Oy
+
+  Derivative Work of Elasticsearch
+  Copyright 2012-2015 Elasticsearch
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+package com.teragrep.lsh_01.conversion;
+
+import jakarta.json.*;
+
+import java.io.StringReader;
+import java.util.List;
+import java.util.function.Function;
+
+/**
+ * A Json array payload splittable into individual json objects.
+ */
+public final class JsonPayload implements Payload, Splittable<Payload> {
+
+    private final String message;
+
+    public JsonPayload(String message) {
+        this.message = message;
+    }
+
+    /**
+     * Splits the array of JSON objects into payloads with one object each. Has a side effect of removing whitespace
+     * from the payloads because of jsonObject.toString().
+     * 
+     * @return split list of payloads
+     */
+    @Override
+    public List<Payload> split() {
```

Comment:
This function is not named correctly right now. The function name represents a manipulator, though it returns a List. Renaming this is quite hard, and even a redesign could be possible?

Reply:
Removed the Splittable interface and changed Payload interface's message() function to messages() that returns a List.